smack-experimental/src/main/java/org/jivesoftware/smackx/message_markup/MessageMarkup.java

```diff
@@ -0,0 +1,220 @@
+/**
+ *
+ * Copyright © 2018 Paul Schaub
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.message_markup;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import org.jivesoftware.smackx.message_markup.element.BlockQuoteElement;
+import org.jivesoftware.smackx.message_markup.element.CodeBlockElement;
+import org.jivesoftware.smackx.message_markup.element.ListElement;
+import org.jivesoftware.smackx.message_markup.element.MarkupElement;
+import org.jivesoftware.smackx.message_markup.element.SpanElement;
+
+public final class MessageMarkup {
+
+    private final List<SpanElement> spans = new ArrayList<>();
+    private final List<BlockQuoteElement> quotes = new ArrayList<>();
+    private final List<CodeBlockElement> codes = new ArrayList<>();
+    private final List<ListElement> lists = new ArrayList<>();
+
+    private MessageMarkup() {
+
+    }
+
+    /**
+     * Return a new Builder for Message Markup elements.
+     * @return builder.
+     */
+    public static MessageMarkup getBuilder() {
+        return new MessageMarkup();
+    }
+
+    /**
+     * Mark a section of a message as deleted.
+     *
+     * @param start start index
+     * @param end end index
+     * @return builder
+     */
+    public MessageMarkup setDeleted(int start, int end) {
```

Comment:
It says "return builder", but it's not part of the build API while I also think it should be. This is true for the following methods too.

Reply:
What do you mean? The `MessageMarkup` class *is* the builder, maybe you missed that?

Reply:
Ahh now I see. Well I had expect it to be called MarkupElementBuilder then, this follows the typical pattern that
1. Builder API type names end with "Builder"
2. If you strip the "Builder" from the type name, you get the type that is build